recipes/cubicinterpolation/all/conanfile.py

```diff
@@ -0,0 +1,98 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class CubicInterpolationConan(ConanFile):
+    name = "cubicinterpolation"
+    homepage = "https://github.com/MaxSac/cubic_interpolation"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Leightweight interpolation library based on boost and eigen."
+    topics = ("interpolation", "splines", "cubic", "bicubic", "boost", "eigen3")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+    exports_sources = ["CMakeLists.txt"]
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def validate(self):
```

Comment:
If your library requires non-header-only boost components, you could also check if boost was built with all the components your lib requires.

There is an interesting pattern in gtsam recipe for such case: https://github.com/conan-io/conan-center-index/blob/master/recipes/gtsam/all/conanfile.py#L121-L128